src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -158,7 +165,12 @@ object PeerDiscoveryManager {
 
   }
 
-  case class DiscoveryNodeInfo(node: Node, addTimestamp: Long)
+
+  sealed abstract class TimedInfo {
+    def addTimestamp: Long
+  }
+  case class DiscoveryNodeInfo(node: Node, addTimestamp: Long) extends TimedInfo
+  case class PingInfo(nodeinfo: DiscoveryNodeInfo, addTimestamp: Long) extends TimedInfo
```

Comment:
Why do we need to store 2 `addTimestamp`s for a node (the one in `PingInfo` and `DiscoveryNodeInfo` in it)? Are we still using the `nodeInfo.addTimestamp`?

Reply:
we are using `nodeInfo.addTimestamp` to sort nodes which responded with pong, and we update this timestamp after their response. Also we scan only most fresh nodes

`pingInfo.addTimestamp` is used to record time of pings for which there can be no answer, and is used to remove oldest unanswered ping.